megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java

```diff
@@ -0,0 +1,3628 @@
+            mapSettings.setBoardSize(setup.getBoardWidth(), setup.getBoardHeight());
+            mapSettings.setBoardsSelectedVector(setup.getBoards());
+            clientgui.getClient().sendMapSettings(mapSettings);
+        } catch (Exception ex) {
+            JOptionPane.showMessageDialog(clientgui.frame, 
+                    "There was a problem while loading the map setup!", "Error", JOptionPane.ERROR_MESSAGE);
+            MegaMek.getLogger().error(ex);
+        }
+    }
+    
+    private void removeBot() {
+        Client c = getSelectedClient();
+        if (!client().bots.containsValue(c)) {
+//        if ((c == null) || c.equals(clientgui.getClient())) {
+//            clientgui.doAlertDialog(Messages.getString("ChatLounge.ImproperCommand"),
+//                    Messages.getString("ChatLounge.SelectBo"));
+            LobbyErrors.showOnlyOwnBot(clientgui.frame);
+            return;
+        }
+        // Delete units first, which safely disembarks and offloads them
+        // Don't delete the bot's forces, as that could also delete other players' entitites
+//        lobbyActions.delete(new ArrayList<>(), game().getPlayerEntities(c.getLocalPlayer(), false), false);
+        c.die();
+        clientgui.getBots().remove(c.getName());
+    }
+    
+    private void doBotSettings() {
+        IPlayer player = playerModel.getPlayerAt(tablePlayers.getSelectedRow());
+        BotClient bot = (BotClient) clientgui.getBots().get(player.getName());
+        BotConfigDialog bcd = new BotConfigDialog(clientgui.frame, bot);
+        bcd.setVisible(true);
+
+        if (bcd.dialogAborted) {
+            return; // user didn't click 'ok', add no bot
+        } else if (bot instanceof Princess) {
+            ((Princess) bot).setBehaviorSettings(bcd.getBehaviorSettings());
+            
+            // bookkeeping:
+            clientgui.getBots().remove(player.getName());
+            bot.setName(bcd.getBotName());
+            clientgui.getBots().put(bot.getName(), bot);
+            player.setName(bcd.getBotName());
+            clientgui.chatlounge.refreshPlayerTable();
+        }
+    }
+    
+    // Put a filter on the files that the user can select the proper file.
+    FileFilter XMLFileFilter = new FileFilter() {
+        @Override
+        public boolean accept(File f) {
+            return (f.getPath().toLowerCase().endsWith(".xml") || f.isDirectory());
+        }
+
+        @Override
+        public String getDescription() {
+            return "Map Setup XML files";
+        }
+    };
+    
+    private void setManualMapWidth() {
+        try {
+            int newMapWidth = Integer.parseInt(fldMapWidth.getText());
+            if (newMapWidth >= 1 && newMapWidth <= 20) {
+                mapSettings.setMapSize(newMapWidth, mapSettings.getMapHeight());
+                clientgui.getClient().sendMapDimensions(mapSettings);
+            }
+        } catch (NumberFormatException e) {
+            // no number, no new map width
```

Comment:
you mean like `boolean tryParse(String source, Object resultValue) {}`? It would help if Java could handle return tupels. (Or anything else C# added in the last few years :-) )